lib/twitter_cldr/formatters/numbers/number_formatter.rb

```diff
@@ -69,24 +69,35 @@ def parse_number(number, options = {})
         precision = options[:precision] || precision_from(number)
         rounding = options[:rounding] || 0
 
-        number = "%.#{precision}f" % round_to(number, precision, rounding).abs
+        if number.is_a? BigDecimal
+          number = precision == 0 ?
```

Comment:
I'm not sure I understand why we need to do anything special if `precision == 0`. I tried `round_to(number, precision, rounding).abs.round(precision).to_s("F")` and all the tests passed. If it's necessary, please add a test that covers the special case.

Reply:
There's a gotcha in Ruby 3.0: precision(0) on a BigDecimal now returns int instead of BigDecimal. This is a change that has Matz's approval, though it is not documented yet. If you're using BigDecimal for arbitrary precision integers, precision(0) in Ruby 3.0 will therefore lose that precision, hence the special processing.

Will add test case.

Reply:
Interesting, thanks for explaining :) While you're working on the additional test case, I'm going to add Ruby 3.0 to the build matrix.

Reply:
Ok, ruby 3.0 has been added and the build is passing. If you merge upstream master, your branch should start to run against ruby 3 as well.

Reply:
Added explicit test for precision(0)